### PR TITLE
Doc markersupdate

### LINKED ISSUE #7342: list of markers in documentation could be improved
The table at the top of http://matplotlib.org/devdocs/api/markers_api.html could be improved.
- [x] The curly quotes are sometimes in the wrong direction (probably just wrap everything in backquotes).
- [ ] What's the difference between "triangle_down" and "tri_down", if any?  It may be better to actually just draw the markers in the table (possibly using some resembling unicode characters, to keep things simpler?).
- [ ] How do I specify TICKLEFT, CARETLEFT and friends?  (and again I have no ideas how these markers look like)
- [x] The various "nothing" entries could be merged together.
- [ ] What does "the size is normalized" mean for `verts`?  Single x/y rescaling so that the max size (between x and y) is equal to whatever?
- [x] The `(numsides, style, angle)` entry could easily fit within the table itself (not sure how easy this is to do with rst though.

### LINKED ISSUE #8556: Add images of markers to the list of markers
Currently http://matplotlib.org/devdocs/api/markers_api.html#module-matplotlib.markers just has marker codes and a description of the marker; it would be nice to have a picture of what each marker looks like as a 3rd column.

### PR DESCRIPTION
**Pull request by ImportanceOfBeingErnest**
## PR Summary

Closes #7342 and closes #8556.

#7342 lists a few things to improve about the [`matplotlib.markers` documenation](https://matplotlib.org/api/markers_api.html?highlight=marker).

This PR fixes those. In particular it

* adds images of the markerstyles to the table
* specifies how to use TICKLEFT, CARETLEFT and friends
* explains what "the size is normalized" means.
* modifies and improves the [marker_reference example](https://matplotlib.org/gallery/lines_bars_and_markers/marker_reference.html)

The marker table will look like

image

<details>
 <summary>For completeness/reproducibility, here is the code to produce the images:</summary>

```
# adapted from my Stackoverflow answer
# https://stackoverflow.com/questions/43021593/how-can-i-export-legend-markers-from-matplotlib-one-at-a-time/43028174#43028174
import matplotlib.pyplot as plt
import numpy as np; np.random.seed(422)
plt.rcParams["legend.labelspacing"] = .7


markers= dict((k,v) for k,v in plt.Line2D.markers.items())
markers.update({"$f$" : "mathtext"})


fig, ax = plt.subplots(dpi=100)
for i, (k,v) in enumerate(markers.items()):
    if v=="nothing": continue
    print("{:02d}".format(i), k,v)
    ax.plot([0,1], [i,i], linestyle="", marker=k, ms=8, label=v, color="C0")

leg = ax.legend(framealpha=1, ncol=3)

def export_legend_handles(fig, leg, filename=None, ext=[".pdf", ".png"], d = [0,0,0,3]):
    """ d = [left, bottom, right, top] pixel to add in the respective dimension """
    import matplotlib.transforms as mtransforms
    boxes = []
    fig.canvas.draw()
    trans = fig.dpi_scale_trans.inverted()
    for vpack in leg._legend_handle_box.get_children():
        for hpack in vpack.get_children():
            drawbox = hpack.get_children()[0]
            w, h, xd, yd = drawbox.get_extent(fig.canvas.get_renderer())
            ox, oy = drawbox.get_offset()  
            pixbox = mtransforms.Bbox.from_bounds(ox-d[0],oy-d[1],w+d[0]+d[2],h+d[1]+d[3])
            inchbox = pixbox.transformed(trans) 
            boxes.append(inchbox)

    filename = filename if filename else __file__[:-3]
    for i, box in enumerate(boxes):
        for ex in ext:
            plt.savefig("{}{:02d}{}".format(filename,i,ex), dpi="figure", bbox_inches=box)

export_legend_handles(fig, leg, filename="m", ext = [".png"], d = [-2,5,-2,5])


def print_mix():
    s = ".. |m{:02d}| image:: /_static/markers/m{:02d}.png"
    for i  in range(42):
        print(s.format(i,i))
print_mix()
plt.show()
```

</details>

## PR Checklist

- [x] Code is PEP 8 compliant
- [x] Documentation is sphinx and numpydoc compliant

### CODE REVIEW

File: lib/matplotlib/markers.py [hunk 1]
```diff
@@ -5,82 +5,149 @@
...
+those cases ``None`` still means 'default'.
+
+Note that special symbols can be defined via the 
+:doc:`STIX math font </tutorials/text/mathtext>`,
+e.g. ``"$\u266B$"``. For an overview over the STIX font symbols refer to the
+`STIX font table <http://www.stixfonts.org/allGlyphs.html>`_.
+Also see the :doc:`/gallery/text_labels_and_annotations/stix_fonts_demo`.
+
+Integer numbers from ``0`` to ``11`` create lines and triangles. Those are
+equally accessible via capitalied variables, like ``CARETDOWNBASE``.
```
Comment (afvincent) [inline lib/matplotlib/markers.py]: capitali*z*ed?

**Comment by ImportanceOfBeingErnest** [comment]
Not sure what the problem of travis-ci is; it looks like it doesn't like itself rather than not liking this PR.

**Comment by ImportanceOfBeingErnest** [comment]
It didn't. Something is generally going wrong with the ci tests currently, other PRs are affected by the same issue.

**Comment by jklymak** [comment]
Merged; I don't think the CI problems are here...

**Comment by choldgraf** [comment]
🎉

File: lib/matplotlib/markers.py [hunk 2]
```diff
@@ -5,82 +5,149 @@
...
+``"8"``                        |m11|  octagon
+``"s"``                        |m12|  square
+``"p"``                        |m13|  pentagon
+``"P"``                        |m23|  plus (filled)
+``"*"``                        |m14|  star
+``"h"``                        |m15|  hexagon1
+``"H"``                        |m16|  hexagon2
+``"+"``                        |m17|  plus
+``"x"``                        |m18|  x
+``"X"``                        |m21|  x (filled)
```
[the remark this answers is not in the document]
Comment (ImportanceOfBeingErnest) [reply lib/matplotlib/markers.py]: Absolutely. Thanks for spotting this. It should be fixed by #11545.

**Pull request merged**